tests/microos/toolbox.pm

```diff
@@ -0,0 +1,88 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Run simple toolbox tests
+# Maintainer: Jose Lausuch <jalausuch@suse.com>
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+use version_utils 'is_sle_micro';
+
+sub run {
+    my ($self) = @_;
+    $self->select_serial_terminal;
+
+    my $user = 'test_user';
+
+    # Create user and don't ask password for sudo commands
+    assert_script_run "useradd -m $user ";
+    assert_script_run "echo \"$user ALL=(ALL) NOPASSWD:ALL\" >> /etc/sudoers";
+
+    # Display help
+    assert_script_run 'toolbox -h';
+
+    record_info 'Test',                    "Run toolbox without flags";
+    assert_script_run 'toolbox -r id',     timeout => 180;
+    validate_script_output 'podman ps -a', sub { m/toolbox-root/ };
+    assert_script_run 'podman rm toolbox-root';
+
+    record_info 'Test',                         "Run toolbox with a given tag";
+    assert_script_run 'toolbox -t test_tag id', timeout => 180;
+    validate_script_output 'podman ps -a',      sub { m/toolbox-root-test_tag/ };
+    assert_script_run 'podman rm toolbox-root-test_tag';
+
+    record_info 'Test', "Run toolbox with a given name";
+    assert_script_run('toolbox -c test_name id');
+    validate_script_output 'podman ps -a', sub { m/test_name/ };
+    assert_script_run 'podman rm test_name';
+
+
+    record_info 'Test', "Rootless toolbox as $user";
+    my $prefix = "runuser -l $user -c";
+    my $uid    = script_output "$prefix 'id -u'";
+    validate_script_output "$prefix 'toolbox -u id'", sub { m/uid=${uid}\(${user}\)/ }, timeout => 180;
+    die "$user shouldn't have access to /etc/passwd!" if (script_run("$prefix 'toolbox -u touch /etc/passwd'") == 0);
+    assert_script_run "$prefix \"podman rm toolbox-$user-user\"";
+
+    record_info 'Test',                                         "Rootfull toolbox as $user";
+    validate_script_output "$prefix 'toolbox -r id'",           sub { m/uid=0\(root\)/ };
+    assert_script_run "$prefix 'toolbox -r touch /etc/passwd'", fail_message => 'Root should have access to /etc/passwd!';
+    assert_script_run "podman rm toolbox-$user";
+
+
+    record_info 'Test', 'Pulling toolbox toolbox -u idimage from different registry';
```

Comment:
neat picking - can we re-phrase it somehow ?

Reply:
Good catch, probably a copy-paste typo.